cmake/modules/unittest.cmake

```diff
@@ -33,7 +33,7 @@ if((NOT DEFINED ZEPHYR_BASE) AND (DEFINED ENV_ZEPHYR_BASE))
   set(ZEPHYR_BASE ${ENV_ZEPHYR_BASE} CACHE PATH "Zephyr base")
 endif()
 
-if(NOT SOURCES)
+if(NOT SOURCES AND EXISTS main.c)
```

Comment:
we should probably deprecate the old style and solely rely on the user to do an `target_sources()`.

This stems back from CMake <3.11, where sources were a required argument to `add_executable()`.

But such cleanup should not prevent this improvement.

Reply:
100% agree, I guess we could add a warning for using `SOURCES` saying that the dev should add it themselves and we'll stop doing that soon.